### ISSUE TITLE
Labels are too close together

### ISSUE THREAD

**Issue by potter2** [UV-CDAT/uvcdat#1339]
As you can see from the attached the isoline labels are too close together. I understand there is a parameter that can be set, but it needs to be available. For the GUI the default also needs to be father apart.
![labels](https://cloud.githubusercontent.com/assets/2963950/7757613/b2f824fa-ffb8-11e4-912b-085365fe1852.png)

**Comment by dlonie** [comment]
Just to update: I'll add some options on the VTK side of things to adjust the label spacing, and @sankhesh and @doutriaux1 will take a look at adding the necessary methods to VCS to expose the option to users.

**Comment by aashish24** [comment]
@dlonie can we get it done ASAP? My understanding is that this should be a fairly easy task that does not require much effort.

**Comment by dlonie** [comment]
"soon"

Still fighting with memory leaks on the refactor :(

**Comment by aashish24** [comment]
Would it be possible to pause that for a moment and get this done so that Sankhesh can add an API to VCS? Or may @sankhesh can do this one (assuming that it would be easy)

**Comment by dlonie** [comment]
I can get it done later today.

**Comment by potter2** [comment]
David,
When you are doing the labels, could to check to see if they work on an overlay (isoline on top of isofill)?

Thanks,
Jerry


On Jul 7, 2015, at 7:40 AM, David Lonie <notifications@github.com<mailto:notifications@github.com>> wrote:


I can get it done later today.

—
Reply to this email directly or view it on GitHub<https://github.com/UV-CDAT/uvcdat/issues/1339#issuecomment-119224515>.

--
Gerald (Jerry) Potter
NASA Goddard Space Flight Center
E-mail  gerald.potter@nasa.gov<mailto:gerald.potter@nasa.gov>
Office  28:S241
Phone    707.315.8314
Analysis and Data Consultant
Department      GSFC:606.2

**Comment by dlonie** [comment]
@potter2 That's a separate issue (#1338). This issue requires work on the VTK side of things, while that one looks to be an issue with primitive ordering on the VCS side.

**Comment by dlonie** [comment]
VTK merge request up: https://gitlab.kitware.com/vtk/vtk/merge_requests/374

After this is accepted into VTK master, I'll cherry pick it onto the UVCDAT VTK fork, and then it will be available to VCS. It defines new methods `vtkLabeledContourMapper::[S|G]etSkipDistance` that define a minimum distance between labels on the same line.

After that, it's up to @doutriaux1 and @sankhesh to add the VCS specific bits to expose it to UV-CDAT users.

**Comment by dlonie** [comment]
Still waiting on the patch to be merged upstream, but I've cherry-picked it to the UVCDAT VTK fork. See my last comment for details on the new API.

**Comment by doutriaux1** [comment]
@potter2 it's in master please try.
